packages/airqo-device-analysis/build/lib/airqloudanalysis/airqoairqloudanalysis/datainitialisation.py

```
@@ -0,0 +1,711 @@
+    all_data = []  # List to store all fetched data
+
+    #converting channel_id to string
+    channel_id = str(channel_id)
+
+    while True:
+        # Generate the URL for the current chunk
+        url = (
+            f"https://thingspeak.com/channels/{channel_id}/feeds.json?"
+            f"start={start_date}T00:00:00Z&end={current_end}T23:59:59Z&api_key={api_key}&results=8000"
+        )
+        response = requests.get(url)
```

Ruff (0.15.12)

[error] Probable use of `requests` call without timeout

(S113)